tests/integration/test_specfile.py

```diff
@@ -684,8 +679,31 @@ def test_update_version(
     assert spec.expanded_version == version
 
 
-def test_trailing_newline(spec_autosetup, spec_no_trailing_newline):
-    spec = Specfile(spec_autosetup)
+def test_trailing_newline(specfile_factory, spec_autosetup, spec_no_trailing_newline):
+    spec = specfile_factory(spec_autosetup)
     assert str(spec)[-1] == "\n"
-    spec = Specfile(spec_no_trailing_newline)
+    spec = specfile_factory(spec_no_trailing_newline)
     assert str(spec)[-1] != "\n"
+
+
+def test_reload(specfile_factory, spec_minimal, spec_traditional):
+    spec = specfile_factory(spec_minimal)
+    before_reload = copy.deepcopy(spec)
+    if spec.path is not None:
+        shutil.copy2(spec_traditional, spec_minimal)
+        spec.reload()
+        after_reload = spec
+    else:
+        spec._file.seek(0)
+        spec._file.truncate()
+        with open(spec_traditional, encoding="utf-8") as f:
+            new_content = f.read()
+        try:
+            spec._file.write(new_content)
+        except TypeError:
+            spec._file.write(new_content.encode("utf-8"))
+        spec._file.flush()
+        spec._file.seek(0)
+        spec.reload()
+        after_reload = spec
+    assert str(before_reload) != str(after_reload)
```

Comment:
This is nice and indeed does test all "modes", however it still doesn't test the case that caused troubles in packit. That of course makes sense only with named files, and their path needs to be removed and replaced with a different file, or, in other words, the path inode has to change.